src/smt/smt_engine.cpp

```diff
@@ -3706,6 +3723,232 @@ Result SmtEngine::assertFormula(const Expr& ex, bool inUnsatCore)
   return quickCheck().asValidityResult();
 }/* SmtEngine::assertFormula() */
 
+/*
+   --------------------------------------------------------------------------
+    Handling SyGuS commands
+   --------------------------------------------------------------------------
+*/
+
+void SmtEngine::declareSygusVar(const std::string& id, Expr var, Type type)
+{
+  d_private->d_sygusVars.push_back(Node::fromExpr(var));
+  Trace("smt") << "SmtEngine::declareSygusVar: " << var << "\n";
+}
+
+void SmtEngine::declareSygusPrimedVar(const std::string& id, Type type)
+{
+#ifdef CVC4_ASSERTIONS
+  d_private->d_sygusPrimedVarTypes.push_back(type);
+#endif
+  Trace("smt") << "SmtEngine::declareSygusPrimedVar: " << id << "\n";
+}
+
+void SmtEngine::declareSygusFunctionVar(const std::string& id,
+                                        Expr var,
+                                        Type type)
+{
+  d_private->d_sygusVars.push_back(Node::fromExpr(var));
```

Comment:
I'm wondering why this is the same vector as in declareSygusVar, does this preserve the previous behavior?

Reply:
Yes, declared functions should be universally quantified.